Core/Core/Modules/Mastery Paths/MasteryPathViewController.swift

```diff
@@ -0,0 +1,230 @@
+            selectedSetID = id
+        }
+    }
+
+    @objc func onSelectedAssignment(_ notification: Notification) {
+        guard let id = notification.userInfo?["id"] as? String,
+            let courseID = masteryPath.moduleItem?.courseID
+        else { return }
+        env.router.route(to: .course(courseID, assignment: id), from: self, options: .detail)
+    }
+}
+
+class MasteryPathAssignmentSetView: UIView {
+    let stackView = UIStackView()
+    let set: MasteryPathAssignmentSet
+
+    init(set: MasteryPathAssignmentSet) {
+        self.set = set
+        super.init(frame: .zero)
+        layer.borderWidth = 1
+        layer.borderColor = UIColor.named(.borderMedium).cgColor
+        layer.cornerRadius = 12
+        backgroundColor = .named(.borderMedium)
+        addSubview(stackView)
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.pin(inside: self)
+        stackView.axis = .vertical
+        stackView.spacing = 0.5
+        for assignment in set.assignments.sorted(by: { $0.position < $1.position }) {
+            let cell = MasteryPathAssignmentCell(assignment: assignment)
+            cell.translatesAutoresizingMaskIntoConstraints = false
+            stackView.addArrangedSubview(cell)
+        }
+        let select = MasteryPathAssignmentSetSelectCell(id: set.id)
+        select.translatesAutoresizingMaskIntoConstraints = false
+        stackView.addArrangedSubview(select)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+class MasteryPathAssignmentCell: UIView {
+    @IBOutlet weak var nameLabel: UILabel!
+    @IBOutlet weak var pointsLabel: UILabel!
+
+    let assignment: MasteryPathAssignment
+
+    init(assignment: MasteryPathAssignment) {
+        self.assignment = assignment
+        super.init(frame: .zero)
+        loadFromXib()
+        backgroundColor = .named(.backgroundLightest)
+        let tap = UITapGestureRecognizer(target: self, action: #selector(onTap(_:)))
+        addGestureRecognizer(tap)
+        accessibilityTraits.insert(.button)
+        NotificationCenter.default.addObserver(self, selector: #selector(onAssignmentSelected(_:)), name: .masteryPathAssignmentSelected, object: nil)
+        update()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func update() {
+        nameLabel.text = assignment.name
+        if let pointsPossible = assignment.pointsPossible?.doubleValue {
+            pointsLabel.isHidden = false
+            pointsLabel.text = String.localizedStringWithFormat(NSLocalizedString("g_points", bundle: .core, comment: ""), pointsPossible)
+        } else {
+            pointsLabel.isHidden = true
+        }
+    }
+
+    @objc func onTap(_ gesture: UITapGestureRecognizer) {
+        NotificationCenter.default.post(name: .masteryPathAssignmentSelected, object: nil, userInfo: ["id": assignment.id])
```

Comment:
Does this have more than 1 observer?

Reply:
Yes.

The cells observe it [here](https://github.com/instructure/canvas-ios/pull/1286/files/214f08d0df168b7d5c39e30c030e1b88304644d0#diff-f3d336d33f7a637fbc77dd6419fb1639R136) so that they can update their selected state (only one cell should be selected at a time)

And the view controller observes it [here](https://github.com/instructure/canvas-ios/pull/1286/files/214f08d0df168b7d5c39e30c030e1b88304644d0#diff-f3d336d33f7a637fbc77dd6419fb1639R53) to show the detail screen.